src/nimy_lisp.nim

```diff
@@ -48,3 +51,85 @@ proc member*[T](el: T; s: openArray[T]): bool =
   result = false
   if s.find(el) >= 0:
     result = true
+
+proc isValid*[T](x: T): bool =
+  when T is bool:
+    result = x
+  elif T is SomeNumber:
+    result = true
+  elif T is array | seq:
+    result = if x.len > 0: true else: false
+  elif (T is proc):
+    result = not x.isNil
+  elif T is ptr | pointer | ref:
+    result = not x.isNil
+  else:
+    # currently not supported
+    when defined(debugWhenLet):
+      echo "Invalid type: ", $(name(T))
+    result = false
+
+macro when_let*(stmts: untyped): untyped =
+  ## Macro similar to Emacs Lisp's `when-let` macro.
+  ## Takes a set of assignments and checks if all of those
+  ## are valid (in elisp if they are `nil`) by calling `isValid`
+  ## on the values being assigned. If all are valid, the body
+  ## after `op:` will be evaluated, otherwise nothing happens.
```

Comment:
I'd like to name this keyword `do:` instead. Taking the earlier `if_let` suggestion, it would look like:

```
if_let:
  ..
  do:
    ..
  else:
    ..
```

WDYT?

Reply:
Scratch that. `do` doesn't work, because it's a keyword and if its' written in the macro we're greeted by 
```
Error: expression expected, but found 'keyword do'
```

Hmm :/